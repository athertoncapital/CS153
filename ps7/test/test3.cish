main() {
  j = 1;
  k = 2;
  g = *(j+12);
  h = k - 1;
  f = g * h;
  e = *(j+8);
  m = *(j+16);
  b = *(f+0);
  c = e + 8;
  d = c;
  k = m + 4;
  j = b;
  return d + j + k;
}
