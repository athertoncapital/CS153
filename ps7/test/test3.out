main:
x0 := $30
x1 := $31
x2 := $16
x3 := $17
x4 := $18
x5 := $19
x6 := $20
x7 := $21
x8 := $22
x9 := $23
j := 1
k := 2
x11 := j+12
x10 := *(x11+0)
g := x10
x12 := k-1
h := x12
x13 := g*h
f := x13
x15 := j+8
x14 := *(x15+0)
e := x14
x17 := j+16
x16 := *(x17+0)
m := x16
x19 := f+0
x18 := *(x19+0)
b := x18
x20 := e+8
c := x20
d := c
x21 := m+4
k := x21
j := b
x22 := d+j
x23 := x22+k
$2 := x23
jump .L0
.L1:
jump .L0
.L0:
$30 := x0
$31 := x1
$16 := x2
$17 := x3
$18 := x4
$19 := x5
$20 := x6
$21 := x7
$22 := x8
$23 := x9
return

graph built
main() {
   j = 1;
   k = 2;
   g = *(j+12);
   h = k-1;
   f = g*h;
   e = *(j+8);
   m = *(j+16);
   b = *(f+0);
   c = e+8;
   d = c;
   k = m+4;
   j = b;
   return d+j+k;
}

graph g {
  b -- c;
  b -- d;
  b -- e;
  b -- k;
  b -- m;
  b -- x0;
  b -- x1;
  b -- x2;
  b -- x20;
  b -- x21;
  b -- x3;
  b -- x4;
  b -- x5;
  b -- x6;
  b -- x7;
  b -- x8;
  b -- x9;
  c -- m;
  c -- x0;
  c -- x1;
  c -- x2;
  c -- x3;
  c -- x4;
  c -- x5;
  c -- x6;
  c -- x7;
  c -- x8;
  c -- x9;
  d -- j;
  d -- k;
  d -- m;
  d -- x0;
  d -- x1;
  d -- x2;
  d -- x21;
  d -- x3;
  d -- x4;
  d -- x5;
  d -- x6;
  d -- x7;
  d -- x8;
  d -- x9;
  e -- f;
  e -- j;
  e -- m;
  e -- x0;
  e -- x1;
  e -- x16;
  e -- x17;
  e -- x18;
  e -- x19;
  e -- x2;
  e -- x3;
  e -- x4;
  e -- x5;
  e -- x6;
  e -- x7;
  e -- x8;
  e -- x9;
  f -- j;
  f -- m;
  f -- x0;
  f -- x1;
  f -- x14;
  f -- x15;
  f -- x16;
  f -- x17;
  f -- x2;
  f -- x3;
  f -- x4;
  f -- x5;
  f -- x6;
  f -- x7;
  f -- x8;
  f -- x9;
  g -- h;
  g -- j;
  g -- k;
  g -- x0;
  g -- x1;
  g -- x12;
  g -- x2;
  g -- x3;
  g -- x4;
  g -- x5;
  g -- x6;
  g -- x7;
  g -- x8;
  g -- x9;
  h -- j;
  h -- x0;
  h -- x1;
  h -- x2;
  h -- x3;
  h -- x4;
  h -- x5;
  h -- x6;
  h -- x7;
  h -- x8;
  h -- x9;
  j -- k;
  j -- x0;
  j -- x1;
  j -- x10;
  j -- x11;
  j -- x12;
  j -- x13;
  j -- x14;
  j -- x15;
  j -- x2;
  j -- x3;
  j -- x4;
  j -- x5;
  j -- x6;
  j -- x7;
  j -- x8;
  j -- x9;
  k -- x0;
  k -- x1;
  k -- x10;
  k -- x11;
  k -- x2;
  k -- x22;
  k -- x3;
  k -- x4;
  k -- x5;
  k -- x6;
  k -- x7;
  k -- x8;
  k -- x9;
  m -- x0;
  m -- x1;
  m -- x18;
  m -- x19;
  m -- x2;
  m -- x20;
  m -- x3;
  m -- x4;
  m -- x5;
  m -- x6;
  m -- x7;
  m -- x8;
  m -- x9;
  x0 -- x1;
  x0 -- x10;
  x0 -- x11;
  x0 -- x12;
  x0 -- x13;
  x0 -- x14;
  x0 -- x15;
  x0 -- x16;
  x0 -- x17;
  x0 -- x18;
  x0 -- x19;
  x0 -- x2;
  x0 -- x20;
  x0 -- x21;
  x0 -- x22;
  x0 -- x23;
  x0 -- x3;
  x0 -- x4;
  x0 -- x5;
  x0 -- x6;
  x0 -- x7;
  x0 -- x8;
  x0 -- x9;
  x1 -- x10;
  x1 -- x11;
  x1 -- x12;
  x1 -- x13;
  x1 -- x14;
  x1 -- x15;
  x1 -- x16;
  x1 -- x17;
  x1 -- x18;
  x1 -- x19;
  x1 -- x2;
  x1 -- x20;
  x1 -- x21;
  x1 -- x22;
  x1 -- x23;
  x1 -- x3;
  x1 -- x4;
  x1 -- x5;
  x1 -- x6;
  x1 -- x7;
  x1 -- x8;
  x1 -- x9;
  x10 -- x2;
  x10 -- x3;
  x10 -- x4;
  x10 -- x5;
  x10 -- x6;
  x10 -- x7;
  x10 -- x8;
  x10 -- x9;
  x11 -- x2;
  x11 -- x3;
  x11 -- x4;
  x11 -- x5;
  x11 -- x6;
  x11 -- x7;
  x11 -- x8;
  x11 -- x9;
  x12 -- x2;
  x12 -- x3;
  x12 -- x4;
  x12 -- x5;
  x12 -- x6;
  x12 -- x7;
  x12 -- x8;
  x12 -- x9;
  x13 -- x2;
  x13 -- x3;
  x13 -- x4;
  x13 -- x5;
  x13 -- x6;
  x13 -- x7;
  x13 -- x8;
  x13 -- x9;
  x14 -- x2;
  x14 -- x3;
  x14 -- x4;
  x14 -- x5;
  x14 -- x6;
  x14 -- x7;
  x14 -- x8;
  x14 -- x9;
  x15 -- x2;
  x15 -- x3;
  x15 -- x4;
  x15 -- x5;
  x15 -- x6;
  x15 -- x7;
  x15 -- x8;
  x15 -- x9;
  x16 -- x2;
  x16 -- x3;
  x16 -- x4;
  x16 -- x5;
  x16 -- x6;
  x16 -- x7;
  x16 -- x8;
  x16 -- x9;
  x17 -- x2;
  x17 -- x3;
  x17 -- x4;
  x17 -- x5;
  x17 -- x6;
  x17 -- x7;
  x17 -- x8;
  x17 -- x9;
  x18 -- x2;
  x18 -- x3;
  x18 -- x4;
  x18 -- x5;
  x18 -- x6;
  x18 -- x7;
  x18 -- x8;
  x18 -- x9;
  x19 -- x2;
  x19 -- x3;
  x19 -- x4;
  x19 -- x5;
  x19 -- x6;
  x19 -- x7;
  x19 -- x8;
  x19 -- x9;
  x2 -- x20;
  x2 -- x21;
  x2 -- x22;
  x2 -- x23;
  x2 -- x3;
  x2 -- x4;
  x2 -- x5;
  x2 -- x6;
  x2 -- x7;
  x2 -- x8;
  x2 -- x9;
  x20 -- x3;
  x20 -- x4;
  x20 -- x5;
  x20 -- x6;
  x20 -- x7;
  x20 -- x8;
  x20 -- x9;
  x21 -- x3;
  x21 -- x4;
  x21 -- x5;
  x21 -- x6;
  x21 -- x7;
  x21 -- x8;
  x21 -- x9;
  x22 -- x3;
  x22 -- x4;
  x22 -- x5;
  x22 -- x6;
  x22 -- x7;
  x22 -- x8;
  x22 -- x9;
  x23 -- x3;
  x23 -- x4;
  x23 -- x5;
  x23 -- x6;
  x23 -- x7;
  x23 -- x8;
  x23 -- x9;
  x3 -- x4;
  x3 -- x5;
  x3 -- x6;
  x3 -- x7;
  x3 -- x8;
  x3 -- x9;
  x4 -- x5;
  x4 -- x6;
  x4 -- x7;
  x4 -- x8;
  x4 -- x9;
  x5 -- x6;
  x5 -- x7;
  x5 -- x8;
  x5 -- x9;
  x6 -- x7;
  x6 -- x8;
  x6 -- x9;
  x7 -- x8;
  x7 -- x9;
  x8 -- x9;
}

